core/src/main/scala/org/apache/spark/metrics/source/StaticSources.scala

```diff
@@ -97,6 +97,12 @@ object HiveCatalogMetrics extends Source {
     MetricRegistry.name("parallelListingJobCount"))
 
   /**
+   * Tracks the total number of cachedDataSourceTables hits.
+   */
+  val METRIC_DATASOUCE_TABLE_CACHE_HITS = metricRegistry.counter(
+    MetricRegistry.name("dataSourceTableCacheHits"))
```

Comment:
I may find the reason of the odd scenario, please check my conclusion:
In 2.0 add a new config lazyPruningEnabled, it's default value is true and while multi-thread do the building same time, it will not do listLeafFile.
So I set the `HIVE_MANAGE_FILESOURCE_PARTITIONS`=false and set the partition larger than `PARALLEL_PARTITION_DISCOVERY_THRESHOLD`(this will cause METRIC_PARALLEL_LISTING_JOB_COUNT +1), test results list below:
without lock:
```
0
0
550   (50 file * 11, 1 is cache.load() and the other 10 is 10 threads)
90
11      (also 1 * 11)
1000
```
and with lock:
```
9
0
100  (2 * 50, 1 is cache.load() and the other 1 is the first threads)
54
2     (also 1 * 2)
550
```
so I can delete the added `dataSourceTableCacheHits` metrics and use `parallelListingJobCount` and `filesDiscovered` instead. Do this in next patch.